automation/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/notebooks/NotebookCustomizationSpec.scala

```diff
@@ -136,5 +136,52 @@ final class NotebookCustomizationSpec extends GPAllocFixtureSpec with ParallelTe
         }
       }
     }
+
+    "should execute user-specified start script" in { billingProject =>
+      implicit val ronToken: AuthToken = ronAuthToken
+
+      withNewGoogleBucket(billingProject) { bucketName =>
+        val ronPetServiceAccount = Sam.user.petServiceAccountEmail(billingProject.value)(ronAuthToken)
+        googleStorageDAO.setBucketAccessControl(bucketName,
+          EmailGcsEntity(GcsEntityTypes.User, ronPetServiceAccount),
+          GcsRoles.Owner)
+
+        // Add the user script to the bucket. This script increments and writes a count to file,
+        // tracking the number of times it has been invoked.
+        val startScriptString = "#!/usr/bin/env bash\n\n" +
+          "count=$(cat ~/leo_test_start_count.txt || echo 0)\n" +
+          "echo $(($count + 1)) > ~/leo_test_start_count.txt"
```

Comment:
Haven't finished the PR yet but does the start script run as `root` or `jupyter-user`? If `root` then `~` won't go to the expected place.

Reply:
Yeah, it's root. Will use $JUPYTER_HOME in that case - should be safe I think